### DISCUSSION TITLE
Query sharding for federated query

### DISCUSSION THREAD

**Discussion by rishabhkumar92** [grafana/mimir#8879] [General]
Hi Team,

We have set up  `-query-frontend.query-sharding-total-shards` as 32 and while max `query-sharding-max-sharded-queries` can still go up to higher number like 128(default).For [federated query](https://grafana.com/docs/mimir/latest/manage/secure/authentication-and-authorization/#grafana-mimir-authentication-and-authorization), we are not seeing more than 32 sharded query.

I was expecting it to more given that we are passing 100 of tenant in header, Can someone confirm if this is a bug or desired behavior?

**How to reproduce?**

- We are on Mimir version 2.12
- Issue a federated query with 100+ tenants and observed sharded query as 32 on QFEs.

**Comment by rishabhkumar92** [comment]
cc: @krajorama

**Reply by 56quarters** [reply]
Federation, running the query for multiple tenants, happens in the queriers after the query-frontend has already sharded and split the query. They're not connected to each other.

**Reply by rishabhkumar92** [reply]
is there a recommended way to optimize performance of federated query further? I saw https://github.com/grafana/mimir/pull/5874 which might help in this case

**Reply by 56quarters** [reply]
That doesn't really make sense. The query-frontend shards queries and splits them by time. Each of those sub-queries are handed to a querier. The querier then performs the query for each tenant.

**Reply by rishabhkumar92** [reply]
yeah I updated the comment above, you mind checking it again.
